src/main/java/com/puppycrawl/tools/checkstyle/checks/javadoc/SummaryJavadocCheck.java

```diff
@@ -384,4 +441,331 @@ private static String trimExcessWhitespaces(String text) {
+    }
+
+    /**
+     * Finds and return if the Inline Tag in description Node has allowed Inline Tags.
+     *
+     * @param description Description Node.
+     * @return true, if first sentence contains allowed tags.
+     */
+    private static boolean checkInlineTagInsideDescription(DetailNode description) {
+        boolean checks = true;
+        for (DetailNode nestedInlineTag : description.getChildren()) {
+            if (nestedInlineTag.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                final DetailNode[] child = nestedInlineTag.getChildren();
+                if (child[1].getType() == JavadocTokenTypes.CUSTOM_NAME
+                        && !ALLOWED_INLINE_TAGS.contains(child[1].getText())) {
+                    checks = false;
+                    break;
+                }
+            }
+        }
+        return checks;
+    }
+
+    /**
+     * Get Inline tag inside Inline tag.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTag(DetailNode... javadoc) {
+        DetailNode tagName = null;
+        if (getInlineTagNode(javadoc) != null) {
+            for (DetailNode child : getInlineTagNode(javadoc).getChildren()) {
+                if (child.getType() == JavadocTokenTypes.CUSTOM_NAME) {
+                    tagName = child;
+                }
+            }
+        }
+        return tagName;
+    }
+
+    /**
+     * Gets InlineTag Node.
+     *
+     * @param javadoc Inline tag DetailNode.
+     * @return DetailNode with CUSTOM_NAME;
+     */
+    private static DetailNode getInlineTagNode(DetailNode... javadoc) {
+        DetailNode tagNode = null;
+        for (DetailNode node : javadoc) {
+            if (node.getType() == JavadocTokenTypes.DESCRIPTION) {
+                for (DetailNode nodeChild : node.getChildren()) {
+                    if (nodeChild.getType() == JavadocTokenTypes.JAVADOC_INLINE_TAG) {
+                        tagNode = nodeChild;
+                    }
+                }
+            }
+        }
+        return tagNode;
+    }
+
+    /**
+     * Checks if period is at the end of sentence.
+     *
+     * @param ast Javadoc root node.
+     * @return violation string
+     */
+    private static String getInlineSummarySentence(DetailNode ast) {
+        final StringBuilder result = new StringBuilder(256);
+        if (getDescriptionNode(ast) != null) {
+            for (DetailNode child : getDescriptionNode(ast).getChildren()) {
+                if (child.getType() == JavadocTokenTypes.TEXT) {
+                    result.append(child.getText());
+                }
+            }
+        }
+
+        if (getInlineTagNodeInsideHtml(ast) != null
+                && getDescriptionNodeInsideHtml(getInlineTagNodeInsideHtml(ast)) != null) {
+            for (DetailNode child
+                    : getDescriptionNodeInsideHtml(getInlineTagNodeInsideHtml(ast)).getChildren()) {
```

Comment:
move `getDescriptionNodeInsideHtml(getInlineTagNodeInsideHtml(ast)` to variable

Reply:
Done